archiver/links.py

```diff
@@ -74,6 +74,21 @@ def validate_links(links):
 
     return list(links)
 
+def new_links(imported_links, existing_links):
+    """
+    Return all links which are in the imported_links but not in the existing_links.
+    This is used to determine which links are new and not indexed jet. Set the
+    ONLY_NEW environment variable to activate this filter mechanism.
+    """
+    new_links = []
+    for i_link in imported_links:
+        found_link_in_existing_links = False
+        for e_link in existing_links:
```

Comment:
This O(n^2) loop will crash for users who have many links (a few people have archived 100k+ links!).

Can you make a dictionary for `existing_links` that maps urls to link objects so this can be an O(n) loop.

Reply:
Good point, I've updated the PR accordingly.